mlflow/models/evaluation/artifacts.py

```diff
@@ -21,3 +30,151 @@ def _save(self, output_artifact_path):
     def _load_content_from_file(self, local_artifact_path):
         self._content = pd.read_csv(local_artifact_path)
         return self._content
+
+
+class ParquetEvaluationArtifact(EvaluationArtifact):
+    def _save(self, output_artifact_path):
+        self._content.to_parquet(output_artifact_path, compression="brotli")
+
+    def _load_content_from_file(self, local_artifact_path):
+        self._content = pd.read_parquet(local_artifact_path)
+        return self._content
+
+
+class NumpyEvaluationArtifact(EvaluationArtifact):
+    def _save(self, output_artifact_path):
+        np.save(output_artifact_path, self._content, allow_pickle=False)
+
+    def _load_content_from_file(self, local_artifact_path):
+        self._content = np.load(local_artifact_path, allow_pickle=False)
+        return self._content
+
+
+class JsonEvaluationArtifact(EvaluationArtifact):
+    def _save(self, output_artifact_path):
+        with open(output_artifact_path, "w") as f:
+            json.dump(self._content, f)
+
+    def _load_content_from_file(self, local_artifact_path):
+        with open(local_artifact_path, "r") as f:
+            self._content = json.load(f)
+        return self._content
+
+
+class TextEvaluationArtifact(EvaluationArtifact):
+    def _save(self, output_artifact_path):
+        with open(output_artifact_path, "w") as f:
+            f.write(self._content)
+
+    def _load_content_from_file(self, local_artifact_path):
+        with open(local_artifact_path, "r") as f:
+            self._content = f.read()
+        return self._content
+
+
+class PickleEvaluationArtifact(EvaluationArtifact):
+    def _save(self, output_artifact_path):
+        with open(output_artifact_path, "wb") as f:
+            pickle.dump(self._content, f)
+
+    def _load_content_from_file(self, local_artifact_path):
+        with open(local_artifact_path, "rb") as f:
+            self._content = pickle.load(f)
+        return self._content
+
+
+EXT_TO_ARTIFACT_MAP = {
+    ".png": ImageEvaluationArtifact,
+    ".jpg": ImageEvaluationArtifact,
+    ".jpeg": ImageEvaluationArtifact,
+    ".json": JsonEvaluationArtifact,
+    ".npy": NumpyEvaluationArtifact,
+    ".csv": CsvEvaluationArtifact,
+    ".parquet": ParquetEvaluationArtifact,
+    ".txt": TextEvaluationArtifact,
+}
+
+TYPE_TO_EXT_MAP = {
```

Comment:
Add _ to make it private. I don't think need to expose it. -> `_TYPE_TO_EXT_MAP`